File: sdks/python/src/boxsdk/models/shared/retentionpolicy_mini.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class RetentionPolicyMiniDispositionActionEnum(str, Enum):
    PERMANENTLY_DELETE = "permanently_delete"
    REMOVE_RETENTION = "remove_retention"

class RetentionPolicyMiniTypeEnum(str, Enum):
    RETENTION_POLICY = "retention_policy"


@dataclass_json
@dataclasses.dataclass
class RetentionPolicyMini:
    r"""RetentionPolicyMini
    A base representation of a retention policy.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: RetentionPolicyMiniTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    disposition_action: Optional[RetentionPolicyMiniDispositionActionEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('disposition_action') }})
    policy_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('policy_name') }})
    retention_length: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retention_length') }})
    

File: java-sdk/lib/src/main/java/models/shared/RetentionPolicyMini.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RetentionPolicyMini
 * A base representation of a retention policy.
**/
public class RetentionPolicyMini {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("disposition_action")
    public RetentionPolicyMiniDispositionActionEnum dispositionAction;
    public RetentionPolicyMini withDispositionAction(RetentionPolicyMiniDispositionActionEnum dispositionAction) {
        this.dispositionAction = dispositionAction;
        return this;
    }
    @JsonProperty("id")
    public String id;
    public RetentionPolicyMini withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("policy_name")
    public String policyName;
    public RetentionPolicyMini withPolicyName(String policyName) {
        this.policyName = policyName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("retention_length")
    public String retentionLength;
    public RetentionPolicyMini withRetentionLength(String retentionLength) {
        this.retentionLength = retentionLength;
        return this;
    }
    @JsonProperty("type")
    public RetentionPolicyMiniTypeEnum type;
    public RetentionPolicyMini withType(RetentionPolicyMiniTypeEnum type) {
        this.type = type;
        return this;
    }
}